iceoryx2-bb/posix/tests/file_lock_tests.rs

```diff
@@ -408,3 +411,44 @@ fn file_lock_write_try_lock_does_not_block() {
         assert_that!(counter.load(Ordering::Relaxed), eq 1);
     });
 }
+
+#[test]
+fn file_lock_read_write_works() {
+    test_requires!(POSIX_SUPPORT_FILE_LOCK);
+
+    let handle = ReadWriteMutexHandle::new();
+    let test = TestFixture::new(&handle);
+    let mut guard = test.sut.write_lock().expect("");
+
+    assert_that!(guard.write(b"hello").unwrap(), eq 5);
+    drop(guard);
+
+    let guard = test.sut.write_lock().expect("");
+
+    let mut content = vec![];
+    assert_that!(guard.read_to_vector(&mut content).unwrap(), eq 5);
+    assert_that!(content, eq b"hello");
+
+    drop(guard);
+
+    let guard = test.sut.read_lock().expect("");
+
+    let mut content = vec![];
+    assert_that!(guard.read_to_vector(&mut content).unwrap(), eq 5);
+    assert_that!(content, eq b"hello");
+}
+
+#[test]
+fn file_lock_try_and_timed_lock_fails_when_locked() {
```

Comment:
food for thought: You could add a blocking test with a watchdog killing the test after the timeout and add the `#[should_panic]` attribute to the tests

Reply:
In this case, the read-write lock detects the deadlock and reports it, so there is no deadlock with the watchdog. But this could be added to the read-write lock test suite ...